airflow/models/baseoperator.py

```diff
@@ -646,6 +646,9 @@ def render_template_from_field(self, attr, content, context, jinja_env):
         rt = self.render_template
         if isinstance(content, str):
             result = jinja_env.from_string(content).render(**context)
+        # Special case for named tuples
+        elif isinstance(content, tuple) and type(content) is not tuple:
+            result = content.__class__(*(rt(attr, e, context) for e in content))
```

Comment:
Now we have a duplicate `isinstance(content, tuple)` check. For clarity, I'd split the (list, tuple) check into this:

```python
elif isinstance(content, list):
    result = [rt(attr, e, context) for e in content]
elif isinstance(content, tuple):
    if type(content) is not tuple:
        # Namedtuple
        result = content.__class__(*(rt(attr, e, context) for e in content))
    else:
        result = [rt(attr, e, context) for e in content]
```

Reply:
That sounds good, but I think that the second `else` should then return a tuple too, since we're distinguishing that case from the list already.